sys/kern/kern_clock.c

```diff
@@ -531,7 +534,10 @@ hardclock(int cnt, int usermode)
 			if (i > 0 && i <= newticks)
 				watchdog_fire();
 		}
-		intr_event_handle(clk_intr_event, NULL);
+#ifdef DEV_ACPI
+		if (!CK_SLIST_EMPTY(&clk_intr_event->ie_handlers))
```

Comment:
Why? intr_event_handle checks CK_SLIST_EMPTY at the start and returns EINVAL if so, with the return value being ignored here.

Reply:
Oh, I'd been detecting it due to #1301.  Not exactly a proper situation though.

Reply:
Says who? I'd argue this is working as intended, and having to peek inside the intr_event to know whether to call intr_event_handle is a bad design that unnecessarily exposes internal details.

Reply:
Perhaps this means the generalization of no handlers meaning stray is wrong?  In which case another flag needs to be added for "no handlers isn't stray"?